root/defaults/nginx/server-confs/authelia-server.conf.sample

```diff
@@ -3,50 +3,31 @@
 
 location ^~ /authelia {
     include /config/nginx/location-confs/*.conf;
-    include /config/nginx/resolver.conf;
-    set $upstream_authelia authelia;
-    proxy_pass http://$upstream_authelia:9091;
+
+    set $upstream_app authelia;
+    set $upstream_port 9091;
+    set $upstream_proto http;
+    proxy_pass $upstream_proto://$upstream_app:$upstream_port;
 }
 
 location = /authelia/api/verify {
     internal;
     if ($request_uri ~ [^a-zA-Z0-9_+-=\!@$%&*?~.:#'\;\(\)\[\]]) {
         return 401;
     }
-    include /config/nginx/resolver.conf;
-    set $upstream_authelia authelia;
-    proxy_pass_request_body off;
-    proxy_pass http://$upstream_authelia:9091;
-    proxy_set_header Content-Length "";
+    include /config/nginx/location-confs/*.conf;
+
+    set $upstream_app authelia;
+    set $upstream_port 9091;
+    set $upstream_proto http;
+    proxy_pass $upstream_proto://$upstream_app:$upstream_port;
 
     # Timeout if the real server is dead
     proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;
 
-    # [REQUIRED] Needed by Authelia to check authorizations of the resource.
-    # Provide either X-Original-URL and X-Forwarded-Proto or
-    # X-Forwarded-Proto, X-Forwarded-Host and X-Forwarded-Uri or both.
-    # Those headers will be used by Authelia to deduce the target url of the user.
-    # Basic Proxy Config
     client_body_buffer_size 128k;
-    proxy_set_header Host $host;
-    proxy_set_header X-Original-URL $scheme://$http_host$request_uri;
-    proxy_set_header X-Real-IP $remote_addr;
-    proxy_set_header X-Forwarded-For $remote_addr;
-    proxy_set_header X-Forwarded-Method $request_method;
-    proxy_set_header X-Forwarded-Proto $scheme;
-    proxy_set_header X-Forwarded-Host $http_host;
-    proxy_set_header X-Forwarded-Uri $request_uri;
-    proxy_set_header X-Forwarded-Ssl on;
-    proxy_redirect  http://  $scheme://;
-    proxy_http_version 1.1;
-    proxy_set_header Connection "";
-    proxy_cache_bypass $cookie_session;
-    proxy_no_cache $cookie_session;
```

Comment:
All of this moved to proxy.conf
We will need to verify this does not cause any issues.

Reply:
Tested, no issues.